compiler/rustc_expand/src/expand.rs

```diff
@@ -990,7 +990,7 @@ pub fn parse_ast_fragment<'a>(
             }
         }
         AstFragmentKind::Ty => AstFragment::Ty(this.parse_ty()?),
-        AstFragmentKind::Pat => AstFragment::Pat(this.parse_pat_allow_top_alt(
+        AstFragmentKind::Pat => AstFragment::Pat(this.parse_pat_allow_top_guard(
```

Comment:
would this change macro behavior, allowing guard patterns when they are previously not accepted as a `$a:pat` before? Add a test for that.

Reply:
done